enhancements/operator-sdk/downstream-operator-sdk.md

```diff
@@ -0,0 +1,398 @@
+  * contains the dependencies for the project
+
+#### Handling Patches
+
+Typically we want to submit all bugs upstream, have them release then sync the
+release downstream. But there will be times we need to fix a problem downstream
+to resolve a bug that we can’t wait for an upstream release. The process here
+will be to keep patches separate to avoid conflicts during syncing.
+
+For fixes to vendored dependencies, have a patch in the patches directory
+that will be applied to the vendor after we run go mod vendor.
+
+For fixes to mainline code, have a patch in the patches directory that will be
+applied before builds. In order to do this, we may have to create a downstream
+`Makefile` that could include the upstream `Makefile` if we need to insert
+patches step.
+
+For either of the above, we will always submit the patches to the appropriate
+upstream repos.
+
+To generate the patches, the easiest approach is to use [`gendiff`][gendiff].
+The general steps would be any file to be changed, copy it with a meaningful
+extension and use `gendiff` to generate the patches to put in the `patch`
+directory.
+
+Let's assume `operator_installer.go` needs a fix downstream to fix a bug.
+
+1. Make backup:
+   * `cp operator_installer.go operator_installer.go.bugXXXXXX`
+1. Make changes required to fix the problem:
+   * `vim operator_installer.go`
+1. Generate patch
+   * cd ROOT_OF_PROJECT
+   * `gendiff directory_of_file .bugXXXXXX > 00_bugXXXXXX.patch`
+1. Add patch to patches directory.
+   * `cp 00_bugXXXXXX.patch patches/`
+   * `git add patches/00_bugXXXXXX.patch`
+   * `git commit -m "Bug XXXXXX: fixing ..."`
+
+Patches will be applied in order. The patches will live in the overlay branch.
+
+### Risks and Mitigation
+
+Moving forward with the automated syncing is a little high risk in the sense
+that there are some unknown issues we may encounter. We will mitigate this risk
+by keeping the existing `openshift/ocp-release-operator-sdk` repo until the
+automated `openshift/operator-sdk` repo is working. Once the cut over is made we
+can remove the `openshift/ocp-release-operator-sdk` repo.
+
+In the event that the automated repo is not ready in time for 4.7, the backup
+plan is to do a 4.7 release of OperatorSDK from the existing
+`openshift/ocp-release-operator-sdk` repo.
+
+## Design Details
+
+### Test Plan
+
+The existing set of Operator SDK upstream tests will be run downstream. Today
+we run tests for the Ansible and Helm operators. A list of the upstream tests we
+would run downstream:
+
+* `test-sanity` # sanity checks like formatting and linters
+* `test-unit`   # unit tests
+* `test-e2e`    # e2e tests
+
+The `test-links` target will *not* be run, since that is used to verify the
+links in the upstream docs.
+
+We will run the tests in the OpenShift CI cluster. We will *not* use
+[kind][kind] downstream. There may need to be some patches maintained downstream
+and/or some changes upstream to ensure the tests run in an OpenShift cluster.
```

Comment:
Just a note on this:

In the past (maybe still now?), we ran one OpenShift cluster per e2e test. If we're going to be syncing more often, we're going to be running CI more often, so we need to see if we can be better stewards of the CI system.

Also, fewer clusters used per PR means decreased likelihood of the CI run failing due to a flake in cluster setup.